### ISSUE TITLE
Become a generic Diagram as Code App

### ISSUE THREAD

**Issue by PHHENS** [SeaDve/Delineate#5]
**Feature summary**
Dear Developper,

Thanks for your work on Dagger.
It would be nice, if it could become a most generic app supporting several
current Diagram as Code Languages : GraphViz, Mermaid, SvgBob, PlantUml ...

Thanks for understanding my very optimistic request ;-)

**Comment by SeaDve** [comment]
I actually plan supporting Mermaid, not sure yet with the others

**Comment by PHHENS** [comment]
Dear Dave,

Thanks for your answer, Mermaid support would be a great step forward.

Personnal, I do not use graphviz, but used mostly :
1- svgbob (wonderfull efficient for quick scribbles)
2- PlantUml (a little more complex, but more customisable that mermaid)
3- Mermaid

I am very excited to check Mermaid support.
Thanks

**Comment by PHHENS** [comment]
Dear Dave,
Take a look at this : https://mbarkhau.github.io/asciigrid/
I think you'll be seduced and give some ideas for delineate ...  ;-)

**Comment by PHHENS** [comment]
Dear Dave,

Manuel, who developed asciigrid, is asking if you plan to allow support a grid based editor :
https://github.com/mbarkhau/asciigrid/issues/4
This could help I to add svgbob support to Delineate.
Can you help him ?

Thanks

**Comment by PHHENS** [comment]
Dear Dave,
I did not have any rust knowledge, but I just recently discovered cargo and crates.io.
I think I pointed something which seems allow to add svgbob support easily :
there is a svgbob crates.io package which seems to have a basic svgbob rendering function...
https://crates.io/crates/svgbob
Could this help you ?
Thanks

**Comment by SeaDve** [comment]
Hi, sorry for the really late reply. I have been really busy for the past few months and must've missed the emails.

> Manuel, who developed asciigrid, is asking if you plan to allow support a grid based editor :

That would be nice, but it will probably need some work.

> Could this help you?

Thanks, that looks useful, though that would also need some work.

**Comment by TheJackiMonster** [comment]
I agree, supporting PlantUml in the future would be great. There seem to be some existing crates for it online. But I'm not sure whether they fit the exact needs.

**Comment by viraj1872** [comment]
Yes, PlantUML support would be very useful indeed. Please consider adding it.

Thanks
